src/mlpack/tests/main_tests/preprocess_string_test.cpp

```diff
@@ -0,0 +1,431 @@
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+#include <mlpack/core.hpp>
+static const std::string testName = "PreprocessString";
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/methods/preprocess/preprocess_string_main.cpp>
+
+#include "test_helper.hpp"
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+using namespace std;
+
+bool CompareFiles(const string& fileName1, const string& fileName2)
+{
+  ifstream f1(fileName1), f2(fileName2);
+  string file1line, file2line;
+  if(!f1.is_open())
+    Log::Fatal << fileName1 << " can't be opened" << endl;
+  if(!f2.is_open())
+    Log::Fatal << fileName2 << " can't be opened" << endl;
+  while(getline(f1, file1line) && getline(f2, file2line))
+  {
+    for (size_t i = 0; i < file1line.length(); i++)
+      if(file1line[i] != file2line[i])
+        return false;
+  }
+  return true;
+}
+
+void CreateFile(vector<vector<string> >& input,
+                const string& filename,
+                const string& columnDelimiter)
+{
+  ofstream file(filename);
+  for (auto& row : input)
+  {
+    for (auto& colum : row)
+    {
+      file << colum << columnDelimiter;
+    }
+    file << "\n";
+  }
+}
+
+vector<vector<string>> stringCleaningInput = {
+  { "5", "MLpaCk. Is!' a FAst! macHInE Learning.", "7", 
+    "MLpaCk. Is!' a FAst! macHInE Learning.", "GsOc. is!' GreAt!." },
+  { "4", "MLpaCk. Is!' a FAst! macHInE Learning.", "9",
+    "MLpaCk. Is!' a FAst! macHInE Learning.", "GsOc. is!' GreAt!." }  
+};
+
+vector<vector<string>> stopwords ={ { "is" }, { "a" } , { "great" } };
+
+struct PreprocessStringTestFixture
+{
+ public:
+  PreprocessStringTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+    CreateFile(stringCleaningInput, "string_test.txt", "\t");
+    CreateFile(stringCleaningInput, "string_test.csv", ",");
+    CreateFile(stopwords, "stopwords.txt", "");
+  }
+
+  ~PreprocessStringTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+    remove("string_test.txt");
+    remove("string_test.csv");
+    remove("stopwords.txt");
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(PreprocessStringMainTest,
+                         PreprocessStringTestFixture);
+/**
+ * Convert lower Case from string input datasetfor txt file.
+ */
+BOOST_AUTO_TEST_CASE(LowerCaseTest)
+{
+  vector<vector<string>> stringLowerInput={
+  { "5", "mlpack. is!' a fast! machine learning.", "7", 
+    "mlpack. is!' a fast! machine learning.", "gsoc. is!' great!." },
+  { "4", "mlpack. is!' a fast! machine learning.", "9",
+    "mlpack. is!' a fast! machine learning.", "gsoc. is!' great!." }  
+  };
+  CreateFile(stringLowerInput,"preprocess_string_lower_test.txt","\t" );
+  SetInputParam("actual_dataset", (std::string) "string_test.txt");
```

Comment:
Could you remove all C-style conversions?
I mean the following:
```c++
(std::string) "string_test.txt"
```
The following code should work:
```c++
SetInputParam<std::string>("actual_dataset", "string_test.txt");
```